aicsimageio/tests/writers/test_timeseries_writer.py

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+from pathlib import Path
+
+import dask.array as da
+import numpy as np
+import pytest
+
+from aicsimageio import exceptions
+from aicsimageio.writers import TimeseriesWriter
+
+
+@pytest.mark.parametrize(
+    "array_creator", [lambda shape: np.random.rand(*shape), da.random.random]
+)
+@pytest.mark.parametrize(
+    "shape, dim_order",
+    [
+        ((30, 100, 100), None),
+        ((30, 100, 100, 3), None),
+        ((30, 100, 100, 4), None),
+        ((100, 30, 100), "XTY"),
+        ((100, 4, 100, 30), "YCXT"),
+        pytest.param(
+            (1, 1),
+            None,
+            marks=pytest.mark.raises(exception=exceptions.InconsistentShapeError),
+        ),
+        pytest.param(
+            (1, 1, 1, 1, 1),
+            None,
+            marks=pytest.mark.raises(exception=exceptions.InconsistentShapeError),
+        ),
+        pytest.param(
+            (1, 1, 1, 1, 1, 1),
+            "STCZYX",
+            marks=pytest.mark.raises(exception=exceptions.InconsistentShapeError),
+        ),
+        pytest.param(
+            (1, 1, 1, 1),
+            "ABCD",
+            marks=pytest.mark.raises(
+                exception=exceptions.InvalidDimensionOrderingError
+            ),
+        ),
+    ],
+)
+@pytest.mark.parametrize("filename", ["a.gif", "b.mp4", "c.mkv", "d.avi"])
+def test_timeseries_writer(tmpdir, array_creator, shape, dim_order, filename):
+    # Create array
+    arr = array_creator(shape)
+
+    # Get save location
+    save_path = Path(tmpdir) / filename
+
+    # Save
+    TimeseriesWriter.save(arr.astype(np.uint8), save_path, dim_order)
+
+    # TODO:
+    # Uncomment this test after reader rework
```

Comment:
just wondering, what reader work is needed here?

Reply:
This is the `DefaultReader` has been failing for a while on timeseries formats.

From Slack a while back:
> I really hate that we support buffers
> TIL DefaultReader has been broken for any of the FFMPEG formats, mp4, mkv, etc
> because imageio.get_reader.get_length() returns np.inf for buffers on FFMPEG formats :upside_down_face:

Reply:
Basically FFMPEG formats can't be read because dask tries to construct a dask array and needs to know how many planes to expect and when it sees `np.inf` it doesn't know how many objects to expect to receive from delayed so crashes.